### PR TITLE
582 approximate samples generates plot output as a side effect

### LINKED ISSUE #582: `approximate-Samples` generates plot output as a side-effect
The plot produced by `approximate` is generated as a side-effect (by `matplot`) when `verbose = TRUE`.  The plot is not part of the function's return value.  This seems sub-optimal (it is difficult for the end user to access the plot object) and makes unit testing more complicated than necessary.

Also, the online doc reads <html>
<body>

verbose | be verbose (progress statements and plot)? (default)
-- | --

</body>
</html>

which suggests that `verbose` is both misnamed and controls two options that should be independent (verbosity and production of the plot).

Suggestion:  
- modify `approximate`'s return value to be a list containing the approximated model and the plot (if requested) rather than just the approximated model
- modify `approximate` so that `verbose` controls only progress statements
- introduce a new parameter to control production of the plot. [`create_plot`, perhaps?]

### PR DESCRIPTION
**Pull request by Puzzled-Face**
Closes #582

### CODE REVIEW

File: examples/Sample-methods-approximate.R [hunk 1]
```diff
@@ -1,34 +1,46 @@
...
 # should be increased for a real case.
 set.seed(94)
-posterior <- approximate(object = samples,
-                         model = model,
-                         data = data,
-                         logNormal=TRUE,
-                         control = list(threshold.stop = 0.1,
-                                        max.time = 1,
-                                        maxit = 1))
+posterior <- (approximate(
```
Comment (danielinteractive) [inline examples/Sample-methods-approximate.R:34]: better do this in 2 steps - first save the `approximate()` result list in a variable, and the subset it. That is much easier to read

File: tests/testthat/test-Samples-methods.R [hunk 2]
```diff
@@ -23,3 +25,135 @@ test_that("names-Samples returns correct names of the parameters", {
   expect_identical(names(samples), c("alpha0", "alpha1", "beta"))
   expect_identical(names(samples2), c("a", "z"))
 })
+
+
+test_that("Approximate fails gracefully with bad input", {
```
Comment (danielinteractive) [inline tests/testthat/test-Samples-methods.R:30]:
```suggestion
test_that("approximate fails gracefully with bad input", {
```

File: tests/testthat/test-Samples-methods.R [hunk 3]
```diff
@@ -23,3 +25,135 @@ test_that("names-Samples returns correct names of the parameters", {
...
+      model = model,
+      data = data,
+      points = c(1:5, "NotNumeric"),
+      control = list(threshold.stop = 0.1, max.time = 1, maxit = 1)
+    ),
+    "Assertion on 'points' failed: Must be of type 'numeric', not 'character'"
+  )
+})
+
+test_that("Approximate works correctly", {
```
Comment (danielinteractive) [inline tests/testthat/test-Samples-methods.R:107]:
```suggestion
test_that("approximate works correctly", {
```

File: tests/testthat/test-Samples-methods.R [hunk 4]
```diff
@@ -23,3 +25,135 @@ test_that("names-Samples returns correct names of the parameters", {
...
+    model = model,
+    data = data,
+    create_plot = FALSE,
+    control = list(threshold.stop = 0.1, max.time = 1, maxit = 1)
+  )
+  expect_equal(length(actual1), 1)
+  expect_set_equal(names(actual1), c("model"))
+})
+
+## nolint end
```
Comment (danielinteractive) [inline tests/testthat/test-Samples-methods.R:159]: since we have cleaned up the tests, please avoid nolint start and end. All of this should be checked by the linter

File: vignettes/example.Rnw [hunk 5]
```diff
@@ -37,7 +37,7 @@
 %% no indentation of paragraphs
 \noindent
 
-<<load, echo=FALSE, results=hide>>=
```
Comment (danielinteractive) [inline vignettes/example.Rnw:40]: I would revert the changes in this file back to the state on the main branch.

**Review by danielinteractive — COMMENTED** [review]
Thanks John! Please see inside few comments.

File: examples/Sample-methods-approximate.R (hunk 1 above)
Comment (Puzzled-Face) [reply examples/Sample-methods-approximate.R:34]: Done

File: tests/testthat/test-Samples-methods.R (hunk 4 above)
Comment (Puzzled-Face) [reply tests/testthat/test-Samples-methods.R:159]: Done.

**Review by danielinteractive — APPROVED** [review]
